lightning/src/ln/channelmanager.rs

```diff
@@ -656,25 +656,74 @@ pub struct ChannelDetails {
 	pub counterparty_features: InitFeatures,
 	/// The value, in satoshis, of this channel as appears in the funding output
 	pub channel_value_satoshis: u64,
+	/// The value, in satoshis, which must always be held in the channel for us. This value ensures
+	/// that if we broadcast a revoked state, our counterparty can punish us by claiming at least
+	/// this value on chain.
+	///
+	/// This value is not included in [`outbound_capacity_msat`] as it can never be spent.
+	///
+	/// This value will be `None` for outbound channels until the counterparty accepts the channel.
+	///
+	/// [`outbound_capacity_msat`]: ChannelDetails::outbound_capacity_msat
+	pub counterparty_selected_channel_reserve_satoshis: Option<u64>,
+	/// The value, in satoshis, which must always be held in the channel for our counterparty. This
+	/// value ensures that if our counterparty broadcasts a revoked state, we can punish them by
+	/// claiming at least this value on chain.
+	///
+	/// This value is not included in [`inbound_capacity_msat`] as it can never be spent.
+	///
+	/// [`inbound_capacity_msat`]: ChannelDetails::inbound_capacity_msat
+	pub holder_selected_channel_reserve_satoshis: u64,
```

Comment:
I think we should avoid "holder" (which isn't well known and may change) and "counterparty" in this interface. Could we use "inbound" and "outbound" to be consistent with the capacity fields? Who selected the value isn't very meaningful to the user.

Reply:
Yea, I agree, but I'm not sure "inbound" and "outbound" is more clear, its a really awkward parameter in that its selected by the opposite party from whom it applies to. Maybe `to_self_min_satoshis` and `to_remote_min_satoshis`?

Reply:
I like `to_self_channel_reserve_satoshis` and `remote_channel_reserve_satoshis`. IMO the fact that it was "selected by the opposite party" just needs a comment, and adding that to the naming is potentially misleading for people not reading carefully.

Reply:
I went with `to_remote_reserve_satoshis` and `to_self_reserve_satoshis`

Reply:
Well I think we're loosing in clarity re-introducing `to_self`/`to_remote` mentions, which have revealed themselves quite error-prone in the past and was the motivation of #633. I agree that "holder" meaning and usage isn't stabilized yet, though there we're just adding more confusions. I believe adopted naming here doesn't resolve the confusion on whom transactions the limits are applied.

What do you think `on_holder_tx_reserve_sats`/`on_counterparty_tx_reserve_sats` ? It's quite matching `on_counterparty_tx_csv` already used in `channelmonitor.rs`.

Reply:
Right, its a bit different using them in the public interface vs internal interfaces, IMO. These values aren't *really* about being on a specific tx, though, they're about person-X-needs-at-least-value-Y. Not sure how to better communicate it.

Reply:
> Yea, I agree, but I'm not sure "inbound" and "outbound" is more clear, its a really awkward parameter in that its selected by the opposite party from whom it applies to. Maybe `to_self_min_satoshis` and `to_remote_min_satoshis`?

As noted, who selected the value isn't very meaningful to the user in this context.

> I went with `to_remote_reserve_satoshis` and `to_self_reserve_satoshis`

Sorry about falling off the conversation, but I don't quite agree with the decision here. If these reserve values are subtracted to produce `inbound_capacity_msat` and `outbound_capacity_msat`, why are "inbound" and "outbound" not appropriate modifiers for them?

And, maybe more to the point, if we are subtracting the reserve values to produce `inbound_capacity_msat` and `outbound_capacity_msat`, why do we need to expose them at all?

Introducing more terminology ("self" and "remote") will lead to more confusion, IMHO. And as @ariard noted, we already decided to stop using these. But now we are reintroducing them but in a different manner than previously used. 😞 

> Right, its a bit different using them in the public interface vs internal interfaces, IMO. These values aren't _really_ about being on a specific tx, though, they're about person-X-needs-at-least-value-Y. Not sure how to better communicate it.

Somewhat related -- but maybe outside the scope of this change -- we are overloading "inbound" and "outbound" for channel creation as seen in the documentation for the reserve fields. Perhaps "acceptor" (i.e. fundee) and "initiator" (i.e. funder) are more appropriate there, which would leave "inbound" and "outbound" as a consistent modifier here for the "side" of the channel. At least for user-facing interfaces? Might need to think about this more.

If "inbound" and "outbound" are only specific  for capacities -- rather than generally for the "side" -- then an alternative, which I believe I proposed awhile ago, is to use no modifier for "our" side and "counterparty" for "their" side.

Reply:
> Sorry about falling off the conversation, but I don't quite agree with the decision here. If these reserve values are subtracted to produce inbound_capacity_msat and outbound_capacity_msat, why are "inbound" and "outbound" not appropriate modifiers for them?

Hmm, that feels off? The capacity (which we could rename to spendable, if relevant) fields are "the amount you can send", here we're talking about a static amount which is always in the channel "to" each side. I'm happy to rename them, but I'm kinda at a loss for what is better here.

> And, maybe more to the point, if we are subtracting the reserve values to produce inbound_capacity_msat and outbound_capacity_msat, why do we need to expose them at all?

Because a user asked for them. I dunno why specifically but you could reasonably imagine decisions around when/which channel to force-close including the reserve values.

> Somewhat related -- but maybe outside the scope of this change -- we are overloading "inbound" and "outbound" for channel creation as seen in the documentation for the reserve fields. Perhaps "acceptor" (i.e. fundee) and "initiator" (i.e. funder) are more appropriate there, which would leave "inbound" and "outbound" as a consistent modifier here for the "side" of the channel. At least for user-facing interfaces? Might need to think about this more.

Happy to do that, that makes sense, I think.

> If "inbound" and "outbound" are only specific for capacities -- rather than generally for the "side" -- then an alternative, which I believe I proposed awhile ago, is to use no modifier for "our" side and "counterparty" for "their" side.

I'm happy with that too, I think, would need to see it in practice I think.

Reply:
> > And, maybe more to the point, if we are subtracting the reserve values to produce inbound_capacity_msat and outbound_capacity_msat, why do we need to expose them at all?
> 
> Because a user asked for them. I dunno why specifically but you could reasonably imagine decisions around when/which channel to force-close including the reserve values.

Shouldn't we find out why before going forward with an interface change? 🙂 Especially if it means re-introducing terminology we decided against using?

The user doesn't always know what they want. We have an opportunity to learn what they are trying to do and how we can better serve them. Let's not let the opportunity pass us by!

Reply:
> Shouldn't we find out why before going forward with an interface change? slightly_smiling_face Especially if it means re-introducing terminology we decided against using?

Heh, fair, I also figured it was a reasonable request and something that could be used programmatically so went ahead with it. Concretely, all of LDK is a programmatic interface, not a human one, so providing more information never really seems like a particularly bad thing to me, as long as the documentation is clear on what its useful for and the variable names are clear. Do you have a concrete rename here?

Reply:
I'd say we should provide interfaces that have a distinct purpose. The naming issues may partly be arising because we are trying to put too much into one interface.

If we limit one interface to routing payments, then much of the naming issues disappear: we would only need to expose one capacity (outbound) and one reserve, so there would be no need to come up with suitable prefixes.

Reply:
I don't think "just split the ChannelDetails into two nearly-identical interfaces" solves the issue? We'll still debate about the use and names of particular variables. Ultimately I really think that the router and the "user" here want the same thing - information about how much money they can send/receive over a channel, as well as whether it is active and usable, so I really don't think two interfaces is "right". I agree we should have specific uses for individual variables, as well as clear documentation that describes how or why a user may wish to use them, though.